lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -1282,6 +1289,7 @@ bool OverrideMatcher::checkOverride(ValueDecl *baseDecl,
 
     // The overridden property must not be mutable.
     if (cast<AbstractStorageDecl>(baseDecl)->supportsMutation() &&
+        attempt != OverrideCheckingAttempt::MismatchedSendability &&
```

Comment:
I suspect that this is a consequence of re-assignment and once that's fixed this would no longer be needed.